client/my-sites/email/email-management/email-home.tsx

```diff
@@ -97,12 +98,21 @@ const EmailHome = ( props: EmailManagementHomeProps ): ReactElement => {
 	const currentRoute = useSelector( ( state ) => getCurrentRoute( state ) );
 	const hasSitesLoaded = useSelector( ( state ) => hasLoadedSites( state ) );
 
-	const { data, isLoading: isSiteDomainLoading } = useGetEmailDomainsQuery(
-		selectedSite?.ID ?? null,
-		{
-			retry: false,
-		}
-	);
+	const {
+		data,
+		isLoading: isSiteDomainLoading,
+		remove: removeEmailDomainsCache,
+	} = useGetEmailDomainsQuery( selectedSite?.ID ?? null, {
+		retry: false,
+	} );
+
+	// Clear the query data when the component unmounts in order to prevent showing stale data when
+	// users return here after e.g. adding a new email forward.
+	useEffect( () => {
+		return () => {
+			removeEmailDomainsCache();
```

Comment:
I recorded two screencasts to illustrate the difference this code makes.

[Before](https://user-images.githubusercontent.com/1101677/171143727-f814bdc4-4fce-4c61-9688-aa1af636ff58.mov)

[After](https://user-images.githubusercontent.com/1101677/171143790-01c1f1e2-ff78-439c-a3e5-585bc65d004c.mov)